code/ship/ship.cpp

```diff
@@ -17811,6 +17810,16 @@ float ArmorType::GetDamage(float damage_applied, int in_damage_type_idx, float d
 				constant_val = 0.0f;
 				curr_arg = constant_val;
 			}
+
+			//face: terrible hack to work consistently with beams and additive damages
+			if (is_beam && adtp->Calculations[i] != AT_TYPE_MULTIPLICATIVE
+				&& adtp->Calculations[i] != AT_TYPE_EXPONENTIAL
+				&& adtp->Calculations[i] != AT_TYPE_EXPONENTIAL_BASE
+				&& adtp->Calculations[i] != AT_TYPE_STORE
+				&& adtp->Calculations[i] != AT_TYPE_LOAD) {
+				curr_arg = curr_arg * (flFrametime * 1000.0f) / i2fl(170);
```

Comment:
Beam damage gets calculated based on `BEAM_DAMAGE_TIME`, but here you've hardcoded the magic number of 170ms. This both inconsistent and poor form.